quantumdb-cli/src/main/java/io/quantumdb/cli/xml/ChangelogLoader.java

```diff
@@ -17,14 +17,21 @@ public Changelog load(Changelog changelog, String file) throws IOException {
 		List<XmlChangeset> changesets = xml.getChangesets();
 
 		Version pointer = changelog.getRoot().getChild();
-		for (XmlChangeset changeset : changesets) {
+		for (int i = 0; i < changesets.size(); i++) {
+			XmlChangeset changeset = changesets.get(i);
 			String changesetId = changeset.getId();
 			if (pointer != null) {
 				for (int index = 1; index <= changeset.getOperations().size(); index++) {
 					XmlOperation<?> xmlOperation = changeset.getOperations().get(index - 1);
 					Operation operation = xmlOperation.toOperation();
 					Operation currentOperation = pointer.getOperation();
 
+					if (pointer.getChangeSet().getAuthor().equals("QuantumDB")) {
+						pointer = pointer.getChild();
+						i--;
```

Comment:
Due to adding a changeset that is not in the xml file, I have to skip this pointer

Reply:
Why not do this before the `for` loop? You only want to potentially do this for the root version right?

Reply:
When you perform the cleanup command, it adds a changeset to the database with the id "cleanup_*randomhash*" by the author QuantumDB, this is at the end of the database metadata table and needs to be filtered because it will not be present in the changelog XML.